contrib/mod_sftp/keys.c

```diff
@@ -5162,7 +5160,7 @@ int sftp_keys_verify_signed_data(pool *p, const char *pubkey_algo,
 #endif /* HAVE_SHA256_OPENSSL */
 
 #if defined(HAVE_SHA512_OPENSSL)
-  } else if (strncmp(sig_type, "rsa-sha2-512", 8) == 0) {
+  } else if (strncmp(sig_type, "rsa-sha2-512", 13) == 0) {
```

Comment:
These two occurrences, using the wrong shorter length in `strncmp(3)` comparisons, were the cause of the bug, I think.